### PR TITLE
Added an AMQP sample to show how to use the extension

### LINKED ISSUE #93: Introduce an axon-example
### Enhancement Description

Following the pattern we have on other extensions (kotlin, mongo, kafka, tracing, etc), we should introduce an `amqp-axon-example` on this repo where it serves to showcase how to use the extension but also for testing purposes when updating it.

### Current Behaviour

There is no example project.

### Wanted Behaviour

N/A

### Possible Workarounds

N/A

### PR DESCRIPTION
**Pull request by lfgcampos**
Closes #93 .

### CODE REVIEW

File: copyright-template.xml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<!--
+  ~ Copyright (c) 2010-2018. Axon Framework
```
Comment (smcvb) [inline copyright-template.xml:2]: This date is outdated.

**Review by smcvb — COMMENTED** [review]
Bunch of minor comments.

File: amqp-axon-example/README.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+## How to run
+
+### Preparation
+
+You will need `docker` and `docker-compose` to run this example.
+
+Please run:
+
+```bash
+docker compose -f ./amqp-axon-example/docker-compose.yaml up -d
```
Comment (corradom) [inline amqp-axon-example/README.md:14]:
```suggestion
docker-compose -f ./amqp-axon-example/docker-compose.yaml up -d
```

File: amqp-axon-example/README.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+Now build the application by running:
+
+```bash
+mvn clean package -f ./amqp-axon-example
+```
+
+### Running example application
+
+You can start the application by running `java -jar ./amqp-axon-example/target/amqp-management-example.jar`.
```
Comment (corradom) [inline amqp-axon-example/README.md:27]:
```suggestion
You can start the application by running `java -jar ./amqp-axon-example/target/amqp-axon-example.jar`.
```

at least for me compiles in that jar file

**Comment by corradom** [comment]
I'm still in doubt if we should call 
**SAMPLE**
over
**EXAMPLE**

I remember that we took a decision for of https://github.com/AxonIQ/code-samples ... maybe we should be consistent in naming.

File: amqp-axon-example/README.md (hunk 2 above)
Comment (lfgcampos) [reply amqp-axon-example/README.md:14]: `docker compose` is part of docker already (not sure if on all SOs), so you don't have to use `docker-compose` anymore.

File: amqp-axon-example/README.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
+# AMQP Axon Springboot Example
+
+This is an example SpringBoot application using the AMQP Axon extension. It uses Axon Server as the Event Store and RabbitMQ as the Message Bus.
```
[the remark this answers is not in the document]
Comment (lfgcampos) [reply amqp-axon-example/README.md]: well, we can.. but for simplicity I guess, our other examples are not using Axon Server and that's why I left it out as well

File: amqp-axon-example/rabbit_enabled_plugins [hunk 5]
```diff
@@ -0,0 +1 @@
+[rabbitmq_management].
```
[the remark this answers is not in the document]
Comment (lfgcampos) [reply amqp-axon-example/rabbit_enabled_plugins:1]: it was a rabbitmq I was playing with, aka leftover =)

File: amqp-axon-example/src/main/kotlin/org/axonframework/extension/amqp/example/client/BankClient.kt [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
+    /**
+     * Creates account once.
+     */
+    @Scheduled(initialDelay = 5_000, fixedDelay = 1000_000_000)
+    fun createAccount() {
+        logger.info { "creating account..." }
+        commandGateway.send<String>(
+            CreateBankAccountCommand(
+                bankAccountId = accountId,
```
[the remark this answers is not in the document]
Comment (lfgcampos) [reply amqp-axon-example/src/main/kotlin/org/axonframework/extension/amqp/example/client/BankClient.kt:52]: I don't know how or why but it doesn't fail even if it uses the same UUID to create an Aggregate. This code is the same on Mongo and I believe it is wrong hence I am updating it here and will update on Mongo Example as well.

File: amqp-axon-example/README.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
+# AMQP Axon Springboot Example
+
+This is an example SpringBoot application using the Axon's AMQP extension. It uses an In Memory Event Store and RabbitMQ as the Message Bus.
```
Comment (smcvb) [inline amqp-axon-example/README.md:3]:
```suggestion
This is an example SpringBoot application using Axon's AMQP extension. It uses an In-Memory Event Store and RabbitMQ as the Message Bus.
```

File: amqp-axon-example/src/main/kotlin/org/axonframework/extension/amqp/example/client/BankClient.kt (hunk 6 above)
[the remark this answers is not in the document]
Comment (smcvb) [reply amqp-axon-example/src/main/kotlin/org/axonframework/extension/amqp/example/client/BankClient.kt:53]: That's interesting. Would warrant a check in the in-mem store maybe :thinking:

File: amqp-axon-example/README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (smcvb) [reply amqp-axon-example/README.md]: True, true. Let's leave it without AxonServer.

File: amqp-axon-example/pom.xml [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-amqp</artifactId>
+            <version>${spring.boot.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.axonframework</groupId>
+            <artifactId>axon-spring-boot-starter</artifactId>
+            <version>${axon.version}</version>
+            <exclusions>
```
[the remark this answers is not in the document]
Comment (smcvb) [reply amqp-axon-example/pom.xml:55]: Just wanted to react to the fact you're removing Axon Server :p

**Review by smcvb — APPROVED** [review]
My concerns have been addressed, hence approving.

**Review by corradom — APPROVED** [review]

**Pull request merged**